src/Language/Embedded/Imperative/Backend/C.hs

```diff
@@ -111,6 +118,12 @@ compArrCMD cmd@(CopyArr arr1 arr2 expl) = do
 compArrCMD (UnsafeFreezeArr (ArrComp arr)) = return $ IArrComp arr
 compArrCMD (UnsafeThawArr (IArrComp arr))  = return $ ArrComp arr
 
+-- | Generates the symbol name as an identifier for a given array.
+newtype BaseArrOf i a = BaseArrOf (Arr i a)
```

Comment:
This wrapper is only needed to provide a `ToIdent` instance for `touchVar`. Maybe if `Arr i a` would be a functor, then `('_':)` could be applied on the identifier inside. If `touchVar` will also be needed on compilation of `CExp` expressions, the same instance will also be required there.